docs/GettingStartedDocs/IntelSDKPortingGuideLinux.md

```diff
@@ -0,0 +1,311 @@
+
+## Migrate ECall/OCall Definitions (EDL Files)
+
+Both Intel SGX SDK for Linux and Open Enclave SDK support the same grammer for
+defining trusted/untrusted functions (aka. ECalls/OCalls) in EDL files.
+However, built-in OCalls are defined in different headers. Intel's built-in
+OCalls are defined in `sgx_tstdc.edl` while Open Enclave SDK's are defined in
+`platform.edl`.
+
+Most EDL files include (by `include` statements) common C headers for both host
+and enclave sides. The most commonly included header is the one defining SGX
+architectural structures, which is `arch.h` in Intel SGX SDK for Linux or
+`openenclave/bits/sgx/sgxtypes.h` in Open Enclave SDK. Please also note that
+some structures may be named differently, e.g., the EINITTOKEN architectural
+structure is defined as `token_t` in Intel SGX SDK for Linux but `einittoken_t`
+in Open Enclave SDK.
+
+The code snippet below shows a way to include/import C headers and EDL
+definitions conditionally, in order to be compatible with both SDKs.
+
+```
+enclave {
+#ifdef OEEDGER8R
+    include "openenclave/bits/sgx/sgxtypes.h";
+    from "openenclave/edl/sgx/platform.edl" import *;
+#else
+    include "arch.h";
+    from "sgx_tstdc.edl" import *;
+#endif
+    /* ECall/OCall definitions go here */
+}
+```
+
+The macro `OEEDGER8R` used above could be defined by appending `-DOEEDGER8R` to
+`oeedger8r` command line. Please note that only `oeedger8r` supports defining
+macros at command line, even though both `sgx_edger8r` and `oeedger8r` support
+preprocessing.
+
+The last thing worth noting is that Open Enclave doesn't support nested ECall
+(i.e., an ECall in the context of an OCall) like the Intel SGX SDK for Linux
+does. Existing enclaves making use of nested ECalls need to be reworked to be
+compatible with the Open Enclave SDK.
+
+## Port C/C++ Source Code
+
+Given similarities in the architectures of both SDKs, there shall not be any
+significant code flow/logic changes required. However, source code
+incompatibilities still exist in
+- Header files - They are structured and/or named differently. Fortunately,
+  Open Enclave provides 2 comprehensive headers, namely `openenclave/enclave.h`
+  and `openenclave/host.h`, to be included by trusted and untrusted code,
+  respectively. A single `#include` should suffice in most cases.
+- APIs - Most Open Enclave APIs are prefixed by `oe_` while Intel's APIs are by
+  `sgx_`. Moreover, some APIs may take parameters in different orders.
+- Structure definitions - Structure members may be named differently. Some
+  structures are organized differently. For example, Intel SGX SDK for Linux
+  defines EINITTOKEN as `token_t` with all MAC'ed fields captured in a child
+  structure `launch_body_t`; while Open Enclave defines it as a flat
+  `einittoken_t` structure.
+- Crypto lib - Intel SGX SDK for Linux supports 2 crypto libs - IPP and
+  OpenSSL, and provides a wrapper layer to unify crypto APIs. Open Enclave
```

Comment:
Thanks a lot for writing this up Cedric!
I think this is a good start.
Will this be expanded to include
1) What customers would do if using the key exchange lib
2) What about features such as sealing
3) Subtle differences in edl support, switchless support, threadbinding , exception handling etc.

Reply:
Thanks and all those are good questions!

In fact, we (Intel) had an internal meeting and we decide to put in more contents/details regarding Intel SGX SDK features that are implemented differently or unimplemented by OE SDK. Just look out for updates to this PR.

Then this doc will become a *real* good start point to be merged into `master`. We will then contribute to OE SDK with new features deemed necessary per customer feedbacks and update this doc accordingly going forward.

Reply:
Thanks! Would the expectation be that you would merge this PR and open new PRs with more details?

Reply:
We are preparing more contents for the last section of the document, such as difference in core APIs, libraries and signing tool, We are ok to either extend this PR or submit new PRs. Merging it right now will make it available for broader review, which is our preference. I'll incorporate the remaining comments and make it ready for merge shortly.